controllers/imageclusterinstall_controller.go

```diff
@@ -781,6 +761,62 @@ func generateInfraID(base string) string {
 	return fmt.Sprintf("%s-%s", base, utilrand.String(5))
 }
 
+// GetValidPullSecret validates the pull secret reference and format, return the pull secret data
+func (r *ImageClusterInstallReconciler) GetValidPullSecret(ctx context.Context, psRef *corev1.LocalObjectReference, namespace string) (string, error) {
+	if psRef == nil || psRef.Name == "" || namespace == "" {
+		return "", fmt.Errorf("missing reference to pull secret")
+	}
+	key := types.NamespacedName{Name: psRef.Name, Namespace: namespace}
+	s := &corev1.Secret{}
+	if err := r.Get(ctx, key, s); err != nil {
+		return "", fmt.Errorf("failed to find secret %s: %v", key.Name, err)
+	}
+	psData, ok := s.Data[corev1.DockerConfigJsonKey]
+	if !ok {
+		return "", fmt.Errorf("secret %s did not contain key %s", key.Name, corev1.DockerConfigJsonKey)
+	}
+
+	err := r.ValidatePullSecret(string(psData))
+	if err != nil {
+		return "", fmt.Errorf("invalid pull secret data in secret %w", err)
+	}
+	return string(psData), nil
+}
+
+// ImagePullSecret checks if the given string is a valid image pull secret and returns an error if not.
+func (r *ImageClusterInstallReconciler) ValidatePullSecret(pullSecret string) error {
+	var s imagePullSecret
+
+	err := json.Unmarshal([]byte(strings.TrimSpace(pullSecret)), &s)
+	if err != nil {
+		return fmt.Errorf("pull secret must be a well-formed JSON: %w", err)
+	}
+
+	if len(s.Auths) == 0 {
+		return fmt.Errorf("pull secret must contain 'auths' JSON-object field")
+	}
+	errs := []error{}
+
+	for d, a := range s.Auths {
+		_, authPresent := a["auth"]
+		_, credsStorePresent := a["credsStore"]
+		if !authPresent && !credsStorePresent {
+			errs = append(errs, fmt.Errorf("invalid pull secret: %q JSON-object requires either 'auth' or 'credsStore' field", d))
+			continue
+		}
+		data, err := base64.StdEncoding.DecodeString(a["auth"].(string))
```

Comment:
Is it valid for this not to be present? Might we end up with a panic here for `a["auth"]` being nil?

Reply:
It's validated few lines above, I'll update the code to use the same var